plugin/src/main/kotlin/com/jraska/module/graph/assertion/ModuleGraphAssertionsPlugin.kt

```diff
@@ -29,13 +29,11 @@ class ModuleGraphAssertionsPlugin : Plugin<Project> {
   override fun apply(project: Project) {
     val graphRules = project.extensions.create(GraphRulesExtension::class.java, Api.EXTENSION_ROOT, GraphRulesExtension::class.java)
 
-    project.afterEvaluate {
```

Comment:
Question: Do you have any idea what Gradle setup causes the scenario where this matters to happen?

I would be keen to understand it and potentially write a test for it 🤔 Thanks

Reply:
I'm taking a look, for now I only found one thing that looks promising. 
After adding the plugin to the android app convention plugin (we have a convention plugin structure similar to NIA), 
the statistics task was generating successfully, but it omitted 3 modules 🤔
Unfortunately I can't share the root level gradle files or the convention plugins, although I think it would help immensely. 
After reading through the #230 issue I even tried appending the --no-configuration-cache param, but it still cannot recognize dependencies. 

My guess is that due to the project structure and size, perhaps some setup in the convention plugins as well, the plugin is only recognizing the module in which gradle file it's been declared - in this case only the root level gradle file, hence this output. 

![Screenshot 2024-08-21 at 12 29 23 PM](https://github.com/user-attachments/assets/62da6f3c-2594-4844-905a-8e213f47e34b)

A funny thing to note is that, before applying this fix, I had to apply the plugin to all the modules to make it work 😅 

Maybe this can give you some hints?

Reply:
What I've essentially done is I've declared the plugin inside the :app module gradle file, since the convention plugin is declared there. 😅 That's why it recognizes some modules, but not all of them ☝️ 
But I don't think this is a good way to go, it definitely makes most sense to be at the root level gradle file. 
However, in special cases like this one, the **projects.afterEvaluate** is invoked upon root level gradle project evaluation, which means other subprojects weren't evaluated by this time. 

My best guess about this is that with complex setups and heavily modularized projects with a looot of modules and perhaps faulty practices (f.ex. our project with a height of 16 🥇), the "afterEvaluate" method simply won't work.
I think that's the reason it works for smaller-scale projects, where it happens so that all subprojects are evaluated by the time the root level **afterEvaluate** is executing.

Reply:
Thanks for all the insights and the PR. Makes sense!

I don't fully undertsand yet, but it feels like the problem comes from some dependncies or even the whole setup being applied after the project with plugin is being evaluated or simply the previous callbacks are called after the tasks are added to Gradle and the graph computed 🤔 all such cases are solved now. 👏 

_Btw I've seen height 64 ;) let me know once you get to number like 5 or 3 - it's entirely possible :)_

Reply:
Hooly f... how is that height even possible 😅 
Yeah, that's our goal - height of at most 5 🤩 

Yes, to me also looks like some synchronicity issue (or intended behavior perhaps), i.e. the root project goes first and then the whole setup get's applied. 🤔 It's sort of like a black box to be honest, I wish there were easy ways (that I know of) to actually debug the Gradle evaluation phases and gain insight into why exactly this is happening. 

But again, even if we got to the bottom of it, it would still be better to go with this change rather than adding a whole new section to the README explaining the do's and don't's of project setup in order to use the plugin. 😅 

Thank you for approval and quick responses! 
And once again thank you for your contributions to Android community with insightful talks and this awesome plugin, cheers! 🍻